lib/fluent/plugin/buffer.rb

```diff
@@ -62,6 +62,72 @@ class BufferChunkOverflowError < BufferError; end # A record size is larger than
         def empty?
           timekey.nil? && tag.nil? && variables.nil?
         end
+
+        def cmp_variables(v1, v2)
+          if v1.nil? && v2.nil?
+            return 0
+          elsif v1.nil? # v2 is non-nil
+            return -1
+          elsif v2.nil? # v1 is non-nil
+            return 1
+          end
+          # both of v1 and v2 are non-nil
+          if v1.keys.sort != v2.keys.sort
```

Comment:
`v1.keys` and `v2.keys` should be assigned to local variables because each `keys` call creates new object.
metadata comparison is called heavily in buffers so reducing object allocation is better.

Reply:
I see. Will fix it.